blog.md

```diff
@@ -0,0 +1,346 @@
+* `TropicalMaxPlus` (also simply called `Tropical`): $(\mathbb{R}, \max, +, -\infty, 0)$;
+* `TropicalMinPlus`: $(\mathbb{R}, \min, +, \infty, 0)$;
+* `TropicalMaxMul`: $(\mathbb{R}^+, \max, \times, 0, 1)$.
+
+In recent years, the tropical numbers have been widely used in various areas, including optimization, physics, and computer science, due to its computational simplicity.
+
+For example, it was shown that solving the groud state energy of spin glass problem can be mapped as contraction of a tropical tensor network, which is actually calculating tons of matrix multiplication of Tropical numbers, as shown in
+* [arxiv: Tropical Tensor Network for Ground States of Spin Glasses](https://arxiv.org/abs/2008.06888)
+* [Lei Wang: "Tropical Tensor Networks"](https://www.youtube.com/watch?v=l_7xZ4trcnE)
+
+Although such reformulation did not actucally reduce the complexity of the problem, but in this way we can fully use the power of parallel computing technology developed in recent years and greatly speed up the computation.
+
+Recently, it has also been considered to use the Tropical algebra in machine learning area, as shown in 
+
+* [Generalizing Backpropagation for Gradient-Based Interpretability](https://arxiv.org/abs/2307.03056)
+* [Torch-Struct: Deep Structured Prediction Library](https://arxiv.org/abs/2002.00876)
+
+For these purposes, a fast implementation of the TropicalGEMM on GPU is on demand.
+
+## Matrix Multiplication of Tropical Numbers on GPU
+
+We achieved fast TropicalGEMM via two packages in `Julia`.
+* [`TropicalNumbers.jl`](github.com/TensorBFS/TropicalNumbers.jl): an interface for Tropical Numbers, which will allow users to use tropical numbers just like normal numbers.
+* [`CuTropicalGEMM.jl`](github.com/TensorBFS/CuTropicalGEMM.jl): a fast implementation of the TropicalGEMM on GPU, fast and easy to use.
+
+In the following part of this section, we will mainly introduce these two packages.
+Including how to use them and how we developed them.
+
+### Julia Programing Language
+
+As we mentioned above, we choose to use the `Julia` programing language, which is becoming more and more popular in the past few years.
+Why we choose `Julia`?
+Let's just copy something from the blog [*Why We Create Julia?*](https://julialang.org/blog/2012/02/why-we-created-julia/):
+
+>We are greedy: we want more.
+>
+>We want a language that's open source, with a liberal license. We want the speed of C with the dynamism of Ruby. We want a language that's homoiconic, with true macros like Lisp, but with obvious, familiar mathematical notation like Matlab. We want something as usable for general programming as Python, as easy for statistics as R, as natural for string processing as Perl, as powerful for linear algebra as Matlab, as good at gluing programs together as the shell. Something that is dirt simple to learn, yet keeps the most serious hackers happy. We want it interactive and we want it compiled.
+
+Briefly, `Julia` is fast, and easy to use.
```

Comment:
```suggestion
Briefly, `Julia` is fast, and easy to use.
```
The paragraph is not sharp enough because it is completely disconnected with the rest.

The best way is to copy the introduction to Julia in a paper (e.g. [the Yao paper](https://arxiv.org/pdf/1912.10877.pdf)):
![image](https://github.com/ArrogantGao/blog_CuTropicalGEMM/assets/6257240/ebfeba2a-62d2-4f55-9087-8b0bb0ae90ab)

Another very good example is [CUDA.jl paper](https://arxiv.org/pdf/1712.03112.pdf):
![image](https://github.com/ArrogantGao/blog_CuTropicalGEMM/assets/6257240/1c94103a-f681-43bf-80f0-b58eacd51235)